add config descriptions per user feedback (#4448)
fishtown-analytics/docs.getdbt.com: website/snippets/_config-description-resource.md

> Resource-specific configurations are applicable to only one dbt resource type rather than multiple resource types. You can define these settings in the project file (`dbt_project.yml`), a property file (`models/properties.yml` for models, similarly for other resources), or within the resource’s file using the `{{ config() }}` macro.

Author: ```suggestion
Resource-specific configurations apply to only one dbt resource type rather than multiple resource types. You can define these settings in the project file (`dbt_project.yml`), a property file (`models/properties.yml` for models, similarly for other resources), or within the resource’s file using the `{{ config() }}` macro. 
```

> Resource-specific configurations are applicable to only one dbt resource type rather than multiple resource types. You can define these settings in the project file (`dbt_project.yml`), a property file (`models/properties.yml` for models, similarly for other resources), or within the resource’s file using the `{{ config() }}` macro.<br /><br /><span>The following resource-specific config code block is only available to {props.meta.resource_type}</span>.

Reviewer: The `{props.meta.resource_type}` part isn't rendering correctly for some reason -- do you know why?

Here are the previews that I'm looking at:

- [models](https://docs-getdbt-com-git-mwong-config-descriptions-dbt-labs.vercel.app/reference/model-configs#available-configurations)
- [seeds](https://docs-getdbt-com-git-mwong-config-descriptions-dbt-labs.vercel.app/reference/seed-configs#available-configurations)
- [snapshots](https://docs-getdbt-com-git-mwong-config-descriptions-dbt-labs.vercel.app/reference/snapshot-configs#available-configurations)
- [sources](https://docs-getdbt-com-git-mwong-config-descriptions-dbt-labs.vercel.app/reference/source-configs#available-configurations)
- [tests](https://docs-getdbt-com-git-mwong-config-descriptions-dbt-labs.vercel.app/reference/test-configs#available-configurations)

<img width="500" alt="image" src="https://github.com/dbt-labs/docs.getdbt.com/assets/44704949/9742021b-a68a-4b8c-b357-6195258d2009">

> Resource-specific configurations are applied to individual dbt resources. They influence specific attributes of the resource, such as [`sql-header`](/reference/resource-configs/sql_header), [`column_types`](/reference/resource-configs/column_types), [`strategy`](/reference/resource-configs/strategy), and so on. You can define these settings in the project file (`dbt_project.yml`), a property file (`models/properties.yml` for models, similarly for other resources), or within the resource’s file using the `{{ config() }}` macro.

Reviewer: Since the code example underneath the snippet shows the _actual_ resource-specific configs (along with hyperlinks) supported by that particular resource type, what would you think about simplifying it to this?

```suggestion
Resource-specific configurations are applicable to only one dbt resource type rather than multiple resource types. You can define these settings in the project file (`dbt_project.yml`), a property file (`models/properties.yml` for models, similarly for other resources), or within the resource’s file using the `{{ config() }}` macro.
```

Resource-specific configurations are applicable to only one dbt resource type rather than multiple resource types. You can define these settings in the project file (`dbt_project.yml`), a property file (`models/properties.yml` for models, similarly for other resources), or within the resource’s file using the `{{ config() }}` macro.<br />

> <span>The following resource-specific config code block is only available to {props.meta.resource_type}</span>:

Reviewer: I'm going approve in a moment, which will trigger the auto-merge.

But adding this to a follow-up PR might read a little easier:

```suggestion
<span>The following resource-specific configurations are only available to {props.meta.resource_type}</span>:
```

Here's how it looks like right now:

<img width="500" alt="image" src="https://github.com/dbt-labs/docs.getdbt.com/assets/44704949/8ee59a57-d92d-4c37-825e-969f30827b24">
